lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -51,7 +51,13 @@ def self.deliver_queue(args, options = {})
   end
 
   def self.deliver(*args)
-    @deliver_mutex.synchronize { deliver_block(*args) }
+    # This was added because in RAILS 5 PUMA sends multiple requests
+    # to the same processa and our DRb server implementation for Automate
+    # Methods is not thread safe. The permit_concurrent_loads  allows the
+    # DRb requests which run in different threads to load symbols.
```

Comment:
@Fryguy @jrafanie 
If there are constants missing Rails has to load them when the Drb thread is running. Without the permit_concurrent_loads the Rails thread would get blocked and the process hangs. So these are not really auto-load, if we had loaded them all upfront we wouldn't have seen this issue, these constants come about when some of the constants are missing and Rails tries to load them.